cmake/OpenCVDetectCUDA.cmake

```diff
@@ -73,7 +73,7 @@ if(CUDA_FOUND)
   elseif(CUDA_GENERATION STREQUAL "Volta")
     set(__cuda_arch_bin "7.0")
   elseif(CUDA_GENERATION STREQUAL "Auto")
-    execute_process( COMMAND "${CUDA_NVCC_EXECUTABLE}" "${OpenCV_SOURCE_DIR}/cmake/checks/OpenCVDetectCudaArch.cu" "--run"
+    execute_process( COMMAND "${CUDA_NVCC_EXECUTABLE}" "-ccbin=${CUDA_HOST_COMPILER}" ${CUDA_NVCC_FLAGS} "${OpenCV_SOURCE_DIR}/cmake/checks/OpenCVDetectCudaArch.cu" "--run"
```

Comment:
Do we need check for empty `CUDA_HOST_COMPILER`?

Perhaps we should do similar thing in similar cases (for example, line 93) for code consistency.

Reply:
@alalek ,

* CUDA_HOST_COMPILER is **always** defined by find_package(CUDA) routines **or** passed by user (case of custom compiler). In short, is **never** empty. I believe we should not check it's emptiness.

Here is a short check using a simple ```message(FATAL_ERROR ${CUDA_HOST_COMPILER})``` debug line when no -DCUDA_HOST_COMPILER was passed to cmake at all :
```
-- CUDA detected: 9.0
CMake Error at cmake/OpenCVDetectCUDA.cmake:76 (message):
  /usr/bin/gcc
```

  * Same PR was acccepted e.g. here: [288 / caffe2](https://github.com/caffe2/caffe2/pull/288)

Reply:
@alalek ,

*  Would like to update also line 94 (the aaarch64 thing ) ? Ok, wait will do it right now.

Reply:
I would also suggest trying to remove FindCUDA.cmake and FindCUDA directory from cmake/ : recent Cmake has up-to-date FindCUDA module - if any issues, they can be solved by updating Cmake. Also newest Cmake has 'native' CUDA compilation which would be probably broken by OpenCV shadow files.